microprofile/config/src/main/java/io/helidon/microprofile/config/FieldSetter.java

```diff
@@ -44,9 +47,15 @@ class FieldSetter {
         if (fieldTypes.field0().rawType().equals(Optional.class)) {
             fieldType = fieldTypes.field1().rawType();
             isOptional = true;
+            isList = false;
+        } else if (Objects.nonNull(fieldTypes.field1()) && List.class.isAssignableFrom(fieldTypes.field0().rawType())) {
```

Comment:
1. must be compared to `List` (same as `Optional` above). If implemented this ways, `LinkedList` would be OK, and that would fail with class cast exception
2. the `field1()` is used above without null check. If the type is null, this should fail (maybe not with NPE, but we do not support untyped `List` or `Optional`)

Reply:
The Optional code was there already, I only indented it.  No other logic changed with that section.

The List check is using Class.isAssignable() and there is a null check on like 51.

Reply:
I was saying you should implement it the same way as the `Optional` is implemented.

Reply:
pushed